### PR TITLE
Obstructed piece two

### PR DESCRIPTION
**Pull request by rachelge13**
This is the obstructed piece method. Works for the queen could work for other pieces as well in the future.

### CODE REVIEW

File: app/models/pieces/queen.rb [hunk 1]
```diff
@@ -3,7 +3,186 @@ def valid_move?(x_destination, y_destination)
     x_dist = (x_coord - x_destination).abs
     y_dist = (y_coord - y_destination).abs
 
+<<<<<<< Updated upstream
```
Comment (kenmazaika) [inline app/models/pieces/queen.rb]: Looks like some merge conflict stuff was left in.

File: app/models/pieces/queen.rb [hunk 2]
```diff
@@ -3,7 +3,186 @@ def valid_move?(x_destination, y_destination)
...
+    else
+    	b = (y_coord..y_destination).to_a
+    end
+
+    @spaces = a.zip(b).to_a
+    @spaces.each do |obs|
+    	unless obs[0] == x_coord && obs[1] == y_coord
+        	return true if game.piece_at(obs[0], obs[1])
+    	end
+=======
```
Comment (kenmazaika) [inline app/models/pieces/queen.rb]: More merge conflict stuff.

File: app/models/pieces/queen.rb [hunk 3]
```diff
@@ -3,7 +3,186 @@ def valid_move?(x_destination, y_destination)
...
+        end
+    when @x_distance == 0 && @y_distance > 0
+      for y_element in (y_coord..y_destination)
+        if game.pieces.where(:x_coord => x_destination, :y_coord => y_element).length > 1
+          return true
+        else
+          return false
+        end
+      end
+>>>>>>> Stashed changes
```
Comment (kenmazaika) [inline app/models/pieces/queen.rb]: More merge conflict stuff.

**Comment by kenmazaika** [comment]
It looks like this commit isn't ready to go because of merge conflict issues.  Maybe we can spend a bit of time on the call this evening to discuss merge conflicts and go over how to fix them.

**Comment by jcrides** [comment]
i'm going to merge this, but i'm also putting a task into trello to move this logic into the piece model so it can be used for all pieces

**Pull request merged**